File: 2、LBOCExpandFriendList/LBExpandFriendList/折叠的表示图/LBExpandCell.m
//
//  LBExpandCell.m
//  LBExpandFriendList
//
//  Created by yunmei on 2018/1/19.
//  Copyright © 2018年 yunmei. All rights reserved.
//

#import "LBExpandCell.h"

@implementation LBExpandCell

+ (instancetype)cellWithTableView:(UITableView *)tableview{
    
    LBExpandCell *cell = [tableview dequeueReusableCellWithIdentifier:@"LBExpandCell"];
    if (!cell) {
        cell = [[LBExpandCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"LBExpandCell"];
    }
    tableview.separatorStyle = UITableViewCellSeparatorStyleNone;
//    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    return cell;
}

-(void)awakeFromNib{
    [super awakeFromNib];
}
-(instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        
        [self setUpLayout];
    }
    return  self;
}
- (void)setUpLayout{
    /// 图标
    _imageView = [[UIImageView alloc]init];
    _imageView.image = [UIImage imageNamed:@"rightArrow"];
    [self.contentView addSubview:_imageView];
    __weak typeof(self.contentView) weakSelf = self.contentView;
    [_imageView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.equalTo(weakSelf);
        make.left.equalTo(weakSelf).offset(33);
        make.width.height.equalTo(@10);
    }];
    /// 标题
    _titleLabel = [[UILabel alloc]init];
    _titleLabel.font = [UIFont systemFontOfSize:14];
    [self.contentView addSubview:_titleLabel];
    [_titleLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.equalTo(weakSelf);
        make.left.mas_equalTo(_imageView.mas_right).offset(25);
    }];
}
@end

File: 3、LBSwiftExpandFriendList/LBSwiftExpandFriendList/可折叠的表示图/LBExpandCell.swift
//
//  LBExpandCell.swift
//  LBSwiftExpandFriendList
//
//  Created by yunmei on 2018/1/19.
//  Copyright © 2018年 liubo. All rights reserved.
//

import UIKit

class LBExpandCell: UITableViewCell {
    lazy var titleLabel:UILabel = UILabel()
    lazy var imageV:UIImageView = UIImageView()
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        setUpLayout()
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}

extension LBExpandCell{
    func setUpLayout(){
        /// 图标
        imageV.image = UIImage.init(named: "rightArrow")
        contentView.addSubview(imageV)
        imageV.snp.makeConstraints { (make) in
            make.centerY.equalTo(contentView);
            make.left.equalTo(contentView).offset(33);
            make.width.height.equalTo(10);
        }
        
        /// 标题
        titleLabel.font = UIFont.systemFont(ofSize: 14)
        contentView.addSubview(titleLabel)
        titleLabel.snp.makeConstraints { (make) in
            make.centerY.equalTo(contentView);
            make.left.equalTo(imageV.snp.right).offset(25);
        }
    }
}
